events/controller.go

```diff
@@ -0,0 +1,72 @@
+package events
+
+import (
+	"context"
+	"fmt"
+	"strings"
+	"time"
+
+	pmetrics "github.com/awslabs/operatorpkg/metrics"
+	"github.com/awslabs/operatorpkg/object"
+	"github.com/samber/lo"
+	v1 "k8s.io/api/core/v1"
+	"k8s.io/apimachinery/pkg/runtime/schema"
+	"k8s.io/utils/clock"
+	controllerruntime "sigs.k8s.io/controller-runtime"
+	"sigs.k8s.io/controller-runtime/pkg/builder"
+	"sigs.k8s.io/controller-runtime/pkg/client"
+	"sigs.k8s.io/controller-runtime/pkg/controller"
+	"sigs.k8s.io/controller-runtime/pkg/manager"
+	"sigs.k8s.io/controller-runtime/pkg/metrics"
+	"sigs.k8s.io/controller-runtime/pkg/predicate"
+	"sigs.k8s.io/controller-runtime/pkg/reconcile"
+)
+
+type Controller[T client.Object] struct {
+	objectGVK  schema.GroupVersionKind
+	startTime  time.Time
+	kubeClient client.Client
+	EventCount pmetrics.CounterMetric
+}
+
+func NewController[T client.Object](client client.Client, clock clock.Clock) *Controller[T] {
+	objectGVK := object.GVK(object.New[T]())
+	EventCountOpts.Subsystem = strings.ToLower(objectGVK.Kind)
```

Comment:
How does this not get overwritten and then shared when multiple event controllers are running?

Reply:
These do get overridden each time we create a new controller, however since the counter is held in the controller struct and created with the object of that kind, we would always be tracking the right counter. Never the less I removed this favor of creating the metrics from a function